src/storage/compression/compression_float.cpp

```diff
@@ -0,0 +1,240 @@
+            metadata.floatMetadata().fac, metadata.floatMetadata().exp);
+        integerEncodedValues[i] = encodedValue;
+    }
+
+    getEncodedFloatBitpacker(metadata)->setValuesFromUncompressed(
+        reinterpret_cast<const uint8_t*>(integerEncodedValues.data()), 0, dstBuffer, dstOffset,
+        numValues, metadata.getChild(BITPACKING_CHILD_IDX), nullMask);
+}
+
+template<std::floating_point T>
+std::unique_ptr<CompressionAlg> FloatCompression<T>::getEncodedFloatBitpacker(
+    const CompressionMetadata& metadata) const {
+    if (metadata.getChild(BITPACKING_CHILD_IDX).isConstant()) {
+        return std::make_unique<ConstantCompression>(getBitpackingLogicalType<T>());
+    } else {
+        return std::make_unique<IntegerBitpacking<EncodedType>>();
+    }
+}
+
+template<std::floating_point T>
+BitpackInfo<typename FloatCompression<T>::EncodedType> FloatCompression<T>::getBitpackInfo(
+    const CompressionMetadata& metadata) {
+    const auto& bitpackMetadata = metadata.getChild(BITPACKING_CHILD_IDX);
+    if (bitpackMetadata.isConstant()) {
+        const auto constValue = bitpackMetadata.min.get<EncodedType>();
+        return {.bitWidth = 0, .hasNegative = (constValue < 0), .offset = constValue};
+    } else {
+        return IntegerBitpacking<EncodedType>::getPackingInfo(bitpackMetadata);
+    }
+}
+
+template<std::floating_point T>
+bool FloatCompression<T>::canUpdateInPlace(std::span<const T> value,
+    const CompressionMetadata& metadata,
+    CompressionMetadata::InPlaceUpdateLocalState& localUpdateState,
+    const std::optional<common::NullMask>& nullMask, uint64_t nullMaskOffset) {
+    size_t newExceptionCount = 0;
+    std::vector<EncodedType> encodedValues(value.size());
+    const auto bitpackingInfo = getBitpackInfo(metadata);
+    const auto& floatMetadata = metadata.floatMetadata();
+    for (size_t i = 0; i < value.size(); ++i) {
+        if (nullMask && nullMask->isNull(nullMaskOffset + i)) {
+            continue;
+        }
+
+        const auto floatValue = value[i];
+        const EncodedType encodedValue =
+            alp::AlpEncode<T>::encode_value(floatValue, floatMetadata.fac, floatMetadata.exp);
+        const double decodedValue =
+            alp::AlpDecode<T>::decode_value(encodedValue, floatMetadata.fac, floatMetadata.exp);
+        if (floatValue != decodedValue) {
+            ++newExceptionCount;
+            encodedValues[i] = bitpackingInfo.offset;
+        } else {
+            encodedValues[i] = encodedValue;
+        }
+    }
+    localUpdateState.floatState.newExceptionCount += newExceptionCount;
+    // TODO(Royi) hack: if we rewrite the entire chunk we don't take into account previous
+    // exceptions
+    const size_t totalExceptionCount =
+        (value.size() == common::StorageConstants::NODE_GROUP_SIZE) ?
+            localUpdateState.floatState.newExceptionCount :
+            floatMetadata.exceptionCount + localUpdateState.floatState.newExceptionCount;
```

Comment:
Currently have this hack since during checkpoint (in Column::checkpointColumnChunkInPlace) we pass in entire column chunks. Thus totalExceptionCount would always be double the actual exception count, making it always greater than exceptionCapacity and so canUpdateInPlace would always return false.

This hack also won't work if we have multiple checkpoint states (e.g. due to insertion + update) as numValues will be split between the checkpoint states.

This won't be a problem anymore once we only pass in updated values during checkpoint so just leaving a TODO here